### DISCUSSION TITLE
Support for TD0 format

### DISCUSSION THREAD

**Discussion by flyman02** [keirf/greaseweazle#321] [Q&A]
HI,
I have a series of images made in TD0 (Teledisk) an old DOS program that is no longer supported.
Among the various formats already present the GW.EXE is there one compatible with TD0?
Thank you
F

**Comment by keirf** [comment]
GW can do IMD which is similar. I would need a specification for TD0 to be able to support it.

**Reply by d235j** [reply]
There is a specification included with ImageDisk. It may not be 100% accurate or complete as the format is proprietary.

Here is a direct link to the file. This link may change.
http://dunfield.classiccmp.org/img42841/td0notes.txt

If this link stops working, it can be accessed from here:
http://dunfield.classiccmp.org/img/index.htm -> Teledisk -> “My notes on .TD0 file”

**Comment by keirf** [comment]
Can you also supply or link to some TD0 images?

**Comment by flyman02** [comment]
I found some links with source code:
https://github.com/jmechnich/wteledsk
https://github.com/durgadas311/teledisk
https://github.com/unterwulf/td0utils

This is an example of TD0 files:
https://dompselaar.org/GEM/www.evmusic.it/turbopage/gem/DEMO1.TD0

**Comment by keirf** [comment]
This is now implemented in master. You can download tip of tree or find a prebuilt artifact here:
https://github.com/keirf/greaseweazle/actions/runs/5170051300

**Comment by flyman02** [comment]
Congratulations nice job.
I did some tests and the reading of the TD compressed files is missing.
I know it is distinguished from the first two bytes "TD" = Normal and "td" = compress.
I read that he uses a Lzss-Huffman algorithm.
Link:
https://github.com/jmechnich/wteledsk/blob/master/src/tdlzhuf.c

If you evening an example:
https://dompselaar.org/GEM/www.evmusic.it/turbopage/gem/MOOG.TD0

**Reply by keirf** [reply]
Thanks, an example was needed. I will look into supporting this.

**Comment by keirf** [comment]
TD compressed file support is now implemented.  You can download tip of tree or find a prebuilt artifact here:
https://github.com/keirf/greaseweazle/actions/runs/5211840955
